local_planner/src/nodes/planner_functions.cpp

```diff
@@ -31,7 +38,14 @@ void filterPointCloud(
           distance = (position - toEigen(xyz)).norm();
           if (distance > min_realsense_dist &&
               distance < histogram_box.radius_) {
-            cropped_cloud.points.push_back(pcl::PointXYZ(xyz.x, xyz.y, xyz.z));
+            // subsampling the cloud
+            PolarPoint p_pol = cartesianToPolar(toEigen(xyz), position);
+            Eigen::Vector2i p_ind = polarToHistogramIndex(p_pol, ALPHA_RES / 2);
+            if (high_res_histogram.get_age(p_ind.y(), p_ind.x()) == 0) {
+              final_cloud.points.push_back(toXYZI(xyz.x, xyz.y, xyz.z, 0));
+              high_res_histogram.set_age(p_ind.y(), p_ind.x(), 1);
```

Comment:
I'm just thinking, now that we store the age in the points, do we still need the age in the histogram? What purpose does it serve anymore?